api/server.py

```diff
@@ -67,6 +68,49 @@ def get_api_token():
 APP = Flask(__name__)
 
 
+# Add the appropriate security headers to all responses
+@APP.after_request
+def add_custom_headers(response):
+    # Cache-Control to manage caching behavior
+    response.headers["Cache-Control"] = "no-store, no-cache, must-revalidate, max-age=0"
+
+    # Content-Security-Policy to prevent XSS and other attacks
+    response.headers["Content-Security-Policy"] = (
+        "default-src 'self'; script-src 'self'; object-src 'none'"
+    )
+
+    # Strict-Transport-Security to enforce HTTPS
+    response.headers["Strict-Transport-Security"] = (
+        "max-age=31536000; includeSubDomains; preload"
+    )
+
+    # X-Content-Type-Options to prevent MIME type sniffing
+    response.headers["X-Content-Type-Options"] = "nosniff"
+
+    # X-Frame-Options to prevent clickjacking
+    response.headers["X-Frame-Options"] = "SAMEORIGIN"
+
+    # Referrer-Policy to control referrer information
+    response.headers["Referrer-Policy"] = "no-referrer-when-downgrade"
+
+    # Permissions-Policy to restrict browser feature access
+    response.headers["Permissions-Policy"] = "geolocation=(), microphone=(), camera=()"
+
+    # X-XSS-Protection to prevent reflected XSS attacks (deprecated in modern browsers)
+    response.headers["X-XSS-Protection"] = "1; mode=block"
+
+    # Expect-CT to enforce Certificate Transparency
+    response.headers["Expect-CT"] = "max-age=86400, enforce"
+
+    # Access-Control-Allow-Origin for CORS
+    response.headers["Access-Control-Allow-Origin"] = "*"
+    response.headers["Access-Control-Allow-Methods"] = "GET, POST, PUT, DELETE, OPTIONS"
```

Comment:
I don't think so, but I didn't remove any from the list because I didn't expect my browser to be `OPTIONS`ing this endpoint, but it did.  I will par it down to `GET`, `POST`, and `OPTIONS` -- your suggestion is smart. 65d5a69fdbb50039d6cf3a86ba6d39a1a0d38f5f